src/rng.rs

```diff
@@ -0,0 +1,374 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+pub use self::implementation::{new, new_from, MainRng};
+#[cfg(any(feature = "mock_base"))]
+pub use self::seed_printer::SeedPrinter;
+#[cfg(any(test, feature = "mock_base"))]
+pub use self::test::Seed;
+
+use rand::Rng;
+
+// `CryptoRng` trait shim.
+// TODO: remove this when we update rand to more recent version as it has its own `CryptoRng` trait.
+pub(crate) trait CryptoRng: Rng {}
+impl<'a, R: CryptoRng> CryptoRng for &'a mut R {}
+impl CryptoRng for MainRng {}
+
+// Note: routing uses different version of the rand crate than threshold_crypto. This is a
+// compatibility adapter between the two.
+pub(crate) struct RngCompat<R>(pub R);
+
+impl<R: Rng> rand_crypto::RngCore for RngCompat<R> {
+    fn next_u32(&mut self) -> u32 {
+        self.0.next_u32()
+    }
+
+    fn next_u64(&mut self) -> u64 {
+        self.0.next_u64()
+    }
+
+    fn fill_bytes(&mut self, dest: &mut [u8]) {
+        self.0.fill_bytes(dest)
+    }
+
+    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), rand_crypto::Error> {
+        self.0.fill_bytes(dest);
+        Ok(())
+    }
+}
+
+impl<R: CryptoRng> rand_crypto::CryptoRng for RngCompat<R> {}
+
+// Rng implementation used in production. Uses `OsRng` for maximum cryptographic security.
+#[cfg(not(any(test, feature = "mock_base")))]
```

Comment:
any reason to not use it for test as well if mock_base is not enabled?

Reply:
I guess lack of reproductibility?

Reply:
I guess I can do that, but what would we gain by it?

Reply:
That would apply to unit tests only and there we should not depend on rngs anyway. But in case we do for some reason, at least it gives us reproducibility, like you said.

Reply:
I was wondering whether we would want test using real random(checking we did not completely messed that) , but I guess unreproductible tests are not that useful.